strawberryfields/engine.py

```diff
@@ -674,12 +668,21 @@ def run(
             xcc.Job(id_=job.id, connection=self.connection).cancel()
             raise KeyboardInterrupt("The job has been cancelled.") from e
 
+        if job.status == "complete":
+            self.log.info(f"The remote job {job.id} has been completed.")
+            return Result(samples=list(job.result.values())[0], is_stateful=False)
+        else:
+            message = f"The remote job {job.id} has failed with status {job.status}."
```

Comment:
Don't forget to include the job metadata in the message since that field contains the details of the error.
```suggestion
            message = f"The remote job {job.id} has failed with status {job.status}: {job.metadata}."
```
(XCC v0.1.1 will introduce the `xcc.Job.metadata` attribute.)

Reply:
Nice, didn't know about this. 💯

Reply:
Let's put this PR on hold until the XCC v0.1.1 is released so that we can access the metadata; I would argue that this is very important debugging information.